### DISCUSSION TITLE
MQTT errors:  Belchertown connects repeatedly then errors AMQJs0008I socket closed

### DISCUSSION THREAD

**Discussion by bkcorwin** [poblabs/weewx-belchertown#819] [Q&A]
Hello, Trying to debug my mqtt websockets configuration with the belchertown skin.  I have mqtt setup and websockets.  Both ports are open and listening.  I can subscribe to the weather topic from the command line and I see data streaming in.  

However, when belchertown connects to the server i get repeated errors that say 

MQTT: MQTT Connected. Subscribing
MQTT: 2022-09-18T20:40:50+01:00: mqtt Connection Lost: AMQJS0008I Socket Closed.

This repeats rapidly and forever.  Note the pi/raspberry user is actually correct and that was the user i setup when I did the mqtt passwd configuration

my weewx.conf belchertown config looks like

````
[[Belchertown]]
        skin = Belchertown
        HTML_ROOT = public_html/belchertown
        [[[Extras]]]
            mqtt_websockets_enabled = 1
            mqtt_websockets_host = 192.168.1.99
            mqtt_websockets_port = 9001
            mqtt_websockets_username = pi
            mqtt_websockets_password = raspberry
            mqtt_websockets_ssl = 0
            mqtt_websockets_topc = weather/loop
````

```
    [[MQTT]]
        server_url = mqtt://pi:raspberry@192.168.1.99:1883/
        topic = weather
        unit_system = US
        binding = archive, loop
        aggregation = aggregate

```

myconfig.conf reads

```
persistence false

allow_anonymous true
password_file /etc/mosquitto/passwd

acl_file /etc/mosquitto/acl

# mqtt
listener 1883
protocol mqtt

# websockets 
listener 9001
protocol websockets
```

What would cause the connection to keep failing like this?

Thanks

**Comment by bkcorwin** [comment]
I note now that when I verify that the port is being listened to, I only see an ipv6 entry for websockets.

`tcp6       0      0 :::9001                 :::*                    LISTEN      498/mosquitto    `

whereas on port 1883

```
tcp        0      0 0.0.0.0:1883            0.0.0.0:*               LISTEN      498/mosquitto       
tcp6       0      0 :::1883                 :::*                    LISTEN      498/mosquitto   
```

Perhaps this is the issue?

**Comment by gary-hammer** [comment]
In your myconfig.conf add this and try.

socket_domain ipv4

**Comment by ThePeePs** [comment]
for your weewx.conf I would change the host for MQTT to be `localhost` if mosquito is on the same host/device as weewx.  also, on your skin config, since you are allowing anonymous access, you don't need to put in the user/password for the websocket, try commenting them out to see if that helps.

Also, seeing mosquito listening "only" on  IPv6 when doing `ss` or `netstat` as `:::9001` means that it's also listening on IPv4 as well.
